pkg/pillar/utils/globalutils.go

```diff
@@ -22,38 +22,32 @@ const (
 
 // EnsureGCFile is used by agents which wait for GlobalConfig to become initialized
 // on startup in order to make sure we have a GlobalConfig file.
-func EnsureGCFile() {
-	pubGlobalConfig, err := pubsub.PublishPersistent("", types.GlobalConfig{})
+func EnsureGCFile(specMap types.ConfigItemSpecMap) {
+	pubGlobalConfig, err := pubsub.PublishPersistent("", types.ConfigItemValueMap{})
 	if err != nil {
 		log.Fatal(err)
 	}
-	ReadAndUpdateGCFile(pubGlobalConfig)
+	ReadAndUpdateGCFile(specMap, pubGlobalConfig)
 }
 
 // ReadAndUpdateGCFile does the work of getting a sane or default
 // GlobalConfig based on the current definition of GlobalConfig which
 // might be different than the file stored on disk if we did an update
 // of EVE.
-func ReadAndUpdateGCFile(pub *pubsub.Publication) {
+func ReadAndUpdateGCFile(specMap types.ConfigItemSpecMap, pub *pubsub.Publication) {
```

Comment:
If we do the changes above - of adding the function "types.NewGlobalConfigSpecMap()" - then you don't need to pass specMap as a parameter here. ReadAndUpdateGCFile can get its instance of specMap using specMap = types.NewGlobalConfigSpecMap()inside the function.

Same is the case with EnsureGCFile..